src/standard-live-connect.js

```diff
@@ -124,69 +124,78 @@ function _getInitializedLiveConnect (liveConnectConfig) {
-    const onPixelPreload = () => emitter.send(C.PRELOAD_PIXEL, '0')
-    const pixelClient = new PixelSender(configWithPrivacy, callHandler, onPixelLoad, onPixelPreload)
-    const resolver = IdentityResolver(postManagedState.data, storageHandler, callHandler)
-    const _push = (...args) => _processArgs(args, pixelClient, postManagedState)
+    const onPixelLoad = () => eventBus.emit(C.PIXEL_SENT_PREFIX, syncContainerData)
+    const onPixelPreload = () => eventBus.emit(C.PRELOAD_PIXEL, '0')
+    const pixelClient = new PixelSender(configWithPrivacy, callHandler, eventBus, onPixelLoad, onPixelPreload)
+    const resolver = IdentityResolver(postManagedState.data, storageHandler, callHandler, eventBus)
+    const _push = (...args) => _processArgs(args, pixelClient, postManagedState, eventBus)
     return {
       push: _push,
       fire: () => _push({}),
       peopleVerifiedId: postManagedState.data.peopleVerifiedId,
       ready: true,
       resolve: resolver.resolve,
       resolutionCallUrl: resolver.getUrl,
-      config: liveConnectConfig
+      config: liveConnectConfig,
+      eventBus: eventBus
     }
   } catch (x) {
     console.error(x)
-    emitter.error('LCConstruction', 'Failed to build LC', x)
+    eventBus.emitErrorWithMessage('LCConstruction', 'Failed to build LC', x)
   }
 }
 
 /**
  * @param {LiveConnectConfiguration} liveConnectConfig
  * @param {StorageHandler} externalStorageHandler
  * @param {CallHandler} externalCallHandler
+ * @param {EventBus} externalEventBus
  * @returns {StandardLiveConnect}
  * @constructor
  */
-export function StandardLiveConnect (liveConnectConfig, externalStorageHandler, externalCallHandler) {
+export function StandardLiveConnect (liveConnectConfig, externalStorageHandler, externalCallHandler, externalEventBus) {
   console.log('Initializing LiveConnect')
+  const configuration = (isObject(liveConnectConfig) && liveConnectConfig) || {}
+  configuration.globalVarName = configuration.globalVarName || 'liQ'
+  const eventBus = externalEventBus || LocalEventBus()
   try {
-    const queue = window.liQ || []
-    const configuration = (isObject(liveConnectConfig) && liveConnectConfig) || {}
-    window && (window.liQ = _getInitializedLiveConnect(configuration) || _standardInitialization(configuration, externalStorageHandler, externalCallHandler) || queue)
+    const queue = window[configuration.globalVarName] || []
+    window && (window[configuration.globalVarName] = _getInitializedLiveConnect(configuration) || _standardInitialization(configuration, externalStorageHandler, externalCallHandler, eventBus) || queue)
     if (isArray(queue)) {
       for (let i = 0; i < queue.length; i++) {
-        window.liQ.push(queue[i])
+        window[configuration.globalVarName].push(queue[i])
       }
     }
+    window.liQ_instances = window.liQ_instances || []
+
+    if (window.liQ_instances.filter(i => i.config.globalVarName === configuration.globalVarName).length === 0) {
```

Comment:
Can we use `filter` here?

Reply:
Do you mean `find`? or `filter` where?

Reply:
Sorry, I meant `find`. @wi101 told me that it does not work in every browser.

Reply:
find should work given that we are using babel. It would be very surprising if it does